src/database/datatools/schedulecompact.py

```
@@ -0,0 +1,127 @@
+        conn.commit()
+    finally:
+        conn.close()
+
+    update_description_schedule_compact(db_path, [schedule_compact_id] + dt[:-2])
+
+def delete_schedulecompact(db_path: str, schedule_compact_id: int) -> None:
+    conn = sqlite3.connect(db_path)
+    cursor = conn.cursor()
+
+    table_name = "schedule_compact"
+    sql = f"DELETE FROM {table_name} WHERE id = ?"
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)